libsignal-service/src/service_address.rs

```diff
@@ -96,6 +96,18 @@ impl ServiceAddress {
         (self.phonenumber.is_some() && self.phonenumber == other.phonenumber)
             || (self.uuid.is_some() && self.uuid == other.uuid)
     }
+
+    /// Returns e164 if present, uuid otherwise.
+    pub fn e164_or_uuid(&self) -> String {
+        if let Some(e164) = self.e164() {
+            return e164;
+        } else if let Some(ref uuid) = self.uuid {
+            return uuid.to_string();
+        }
+        unreachable!(
+            "an address requires either a UUID or a E164 phone number"
```

Comment:
technically since `ServiceAddress` is public, you could build it in an invalid state... do we really want this? (I know we have a few panics here and there in the lib, but maybe we can try avoiding new ones?).

Reply:
Hmm, then I'd rather have the initializer set to private at some point in the future.